docs: add how-to guide about linking to external references using Intersphinx (#565)
canonical/sphinx-docs-starter-pack: docs/how-to/optional-customisation/external-referencing-intersphinx.rst

.. meta::
   :description: Configure references to external Sphinx projects with the Intersphinx extension.

.. _how-to-link-docs-intersphinx:

> Link to other documentation sets with Intersphinx

Reviewer: Great title!

=================================================

This guide provides instructions on how to set up external references using the Intersphinx extension. 
Linking to external Sphinx-based documentation sets via hyperlinks is difficult to maintain. 
Intersphinx provides a more robust alternative, linking to sections of other Sphinx projects based on 

> labels mapped out in an inventory list ``objects.inv``. To effectively use Intersphinx, sections in 

Reviewer: Nit:

```suggestion
labels mapped out in an inventory list as ``objects.inv``. To effectively use Intersphinx, sections in 
```

the target documentation set must have labels.

Configure the extension
-----------------------

In the ``conf.py`` file in your docs directory, add or enable ``sphinx.ext.intersphinx`` under ``extensions``:

.. code-block:: python
   :caption: conf.py

   extensions = [
       ...
       "sphinx.ext.intersphinx",
       ]

> In the ``conf.py`` file, connect to the other project in the ``intersphinx_mapping``

Reviewer: Nit:

```suggestion
Then, connect to the other project in the ``intersphinx_mapping``
```

setting:

.. code-block:: python
   :caption: conf.py

   # Add configuration for intersphinx mapping
   # Map only the Sphinx documentation sets that you need to link to from your docs set.
   intersphinx_mapping = {
       'project-key': ('https://example.com', None)
   }

Replace ``project-key`` with the internal identifier for the project that you'll specify
in the document markup, for example ``:external+chisel:ref:`target```. The value
after the link points to a custom path location. If a project stores its ``objects.inv``

> at a special location, replace ``None`` with the path to it.

Reviewer: Recommended:

```suggestion
Replace ``project-key`` with the internal identifier for the project that you'll specify
in the document markup. For example, if the identifier were ``chisel``, an Intersphinx link to it would be ``:external+chisel:ref:`target```.

The value after the URI points to a custom path location. If a project stores its ``objects.inv`` at a special location, replace ``None`` with the path to it.
```

This should help clarify and remove any doubt between what we mean by "link".

Check the external target labels
--------------------------------

To check external target page labels, either search the project source code 
manually or inspect the ``objects.inv`` file. In this file, each section has a 

> unique label. To access the file, run:

Reviewer: Recommended:

```suggestion
unique label.

To inspect the file manually, run:
```

.. code-block:: bash

> 

Reviewer: Oh, I forgot. This command won't work in a plain environment. The user will need to source the venv first:

```suggestion

    source .sphinx/venv/bin/activate
```
